arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -1124,3 +1262,7 @@ inline fun <A, B> EitherOf<A, B>.handleErrorWith(f: (A) -> EitherOf<A, B>): Eith
       is Right -> it
     }
   }
+
+@PublishedApi
+internal suspend inline fun <A> handleItSafely(f: suspend () -> Either<Throwable, A>): Either<Throwable, A> =
```

Comment:
Can this be another overload of Either.catch and become public?

Reply:
With an overload like:
```
    @JvmName("catch1")
    suspend inline fun <R> catch(f: suspend () -> Either<Throwable, R>): Either<Throwable, R> =
      catch<Either<Throwable, R>>(f).fold({ it.left() }, { it })
```
The compiler gets confused: _Type mismatch: inferred type is Int but Either<Throwable, Int> was expected_ (This compile error message is for the "catch should return Right(result) when f does not throw" test case.).

Therefore, I choose to not overload but go with a different function name; `catchAndFlatten`. But the function is now more similar to the `catch` function and is public.

Reply:
Sounds good, thanks!